src/Faker/Name.cs

```diff
@@ -46,6 +46,15 @@ public static class Name
 		/// <returns>The randomly created first name.</returns>
 		public static string First()
 		{
+			return Resources.Name.First.Split(Config.SEPARATOR).Random().Trim();
+		}
+
+		/// <summary>
+		///   Creates a random first name, using a more aggressive caching strategy.
+		/// </summary>
+		/// <returns>The randomly created first name.</returns>
+		public static string FirstCached()
```

Comment:
This would be better suited to be an overload for the First method with a boolean value to the if it should use the cache or not.